### PR TITLE
enable Intrptrcast by default

### PR DESCRIPTION
**Pull request by RalfJung**
As laid out in https://github.com/rust-lang/miri/issues/785: we change Miri to always have an RNG, seeded per default with 0. Then we adjust everything to remove dead code and dead tests.

r? @oli-obk
Cc @christianpoveda

### CODE REVIEW

File: tests/run-pass/bitop-beyond-alignment.rs [hunk 1]
```diff
@@ -33,5 +33,5 @@ fn is_u64_aligned(u: &Tag<u64>) -> bool {
 
 pub fn main() {
     let x = mk_rec();
-    assert!(is_u64_aligned(&x.t));
+    is_u64_aligned(&x.t); // the result of this is non-deterministic
```
Comment (oli-obk) [inline tests/run-pass/bitop-beyond-alignment.rs]: with a specific seed it would be deterministic, right?

**Comment by oli-obk** [comment]
r=me with the comment updated to explain what is meant by "nondeterministic"

File: tests/run-pass/bitop-beyond-alignment.rs (hunk 1 above)
Comment (RalfJung) [reply tests/run-pass/bitop-beyond-alignment.rs]: Yes -- per platform. Results still vary between Linux and macOS, for example.

**Comment by RalfJung** [comment]
Network failure.

@bors retry

**Pull request merged**